### PR TITLE
Limit max-asserts errors to 1 per test (fixes #45)

### LINKED ISSUE #45: max-assertions warns for every additional assertion.
If you use `max-assertions`, it warns for every additional assertion. I think it should just warn for the first assertion in excess of the max (at most once per test). Perhaps the message could specifically say how far over the limit.

### PR DESCRIPTION
**Pull request by jfmengels**
Limits max-asserts errors to 1 per test (fixes #45)

### CODE REVIEW

**Comment by jfmengels** [comment]
> @jamestalmage: Perhaps the message could specifically say how far over the limit.

Forgot this part which I think will be useful. Will work on it right away, WIP in the meantime.

File: rules/max-asserts.js [hunk 1]
```diff
@@ -23,15 +24,19 @@ module.exports = function (context) {
...
+				if (assertionCount === maxAssertions + 1) {
+					nodeToReport = node;
 				}
 			}
 		},
 		'CallExpression:exit': function (node) {
 			if (ava.currentTestNode === node) {
 				// leaving test function
+				if (assertionCount > maxAssertions) {
+					context.report(nodeToReport, 'Expected up to ' + maxAssertions + ' assertions but found ' + assertionCount);
```
Comment (sindresorhus) [inline rules/max-asserts.js]: `assertions but` => `assertions, but`?

**Comment by jfmengels** [comment]
Changed the wording of the error from `Too many assertions` to `Expected up to X assertions but found Y`.

File: rules/max-asserts.js (hunk 1 above)
Comment (jfmengels) [inline rules/max-asserts.js]: :+1: Done

File: rules/max-asserts.js [hunk 2]
```diff
@@ -23,15 +24,19 @@ module.exports = function (context) {
...
+				if (assertionCount === maxAssertions + 1) {
+					nodeToReport = node;
 				}
 			}
 		},
 		'CallExpression:exit': function (node) {
 			if (ava.currentTestNode === node) {
 				// leaving test function
+				if (assertionCount > maxAssertions) {
+					context.report(nodeToReport, 'Expected up to ' + maxAssertions + ' assertions, but found ' + assertionCount);
```
Comment (jamestalmage) [inline rules/max-asserts.js]: `up to` => `at most`

Comment (jfmengels) [inline rules/max-asserts.js]: Sounds good to me, will change it. Would you mind explaining the difference though?

Comment (jfmengels) [inline rules/max-asserts.js]: Done

Comment (jamestalmage) [inline rules/max-asserts.js]: both work. `at most` just sounds better to my ears.

Comment (jfmengels) [inline rules/max-asserts.js]: Ok. Wasn't sure I missed a subtelty ;)

**Comment by sindresorhus** [comment]
Perfect :)

**Comment by jfmengels** [comment]
:)

**Pull request merged**